### DISCUSSION TITLE
Android 平台上 process_connections 出现内存地址不对齐（BUS_ADRALN）的crash

### DISCUSSION THREAD

**Discussion by stopandjoin** [Tencent/tquic#299] [Q&A (问题和回答)]
F DEBUG   : signal 7 (SIGBUS), code 1 (BUS_ADRALN), fault addr 0x2aaab4dd654a
F DEBUG   :     x0  00003ffeff028500  x1  00003ffef1b14800  x2  00002aaab4dd654a  x3  0000000000000002
F DEBUG   :     x4  000000000000018e  x5  00003ffeff4baa88  x6  00003ffeff18e4e0  x7  00003ffeff18e620
F DEBUG   :     x8  00002aaab4ec9900  x9  0000000000000000  x10 00003ffefe7d5e30  x11 0000000000000000
F DEBUG   :     x12 00003ffeff18e640  x13 00003ffef23a2d60  x14 0000000000000000  x15 00000000000001b8
F DEBUG   :     x16 00003fff7fcf0bd8  x17 00003fff7fc83088  x18 0000000000000001  x19 0000000000000010
F DEBUG   :     x20 00003ffeff0521c0  x21 0000000000000010  x22 000000000000120e  x23 00002aaab4b6f11c
F DEBUG   :     x24 00000000000001b8  x25 00003ffefe6d9000  x26 517cc1b727220a95  x27 00003ffefe18e460
F DEBUG   :     x28 00003ffefe18e460  x29 00003ffefe7d6340
F DEBUG   :     sp  00003ffefe7d5d00  lr  00002aaab4cf818c  pc  00002aaab4dd654a
F DEBUG   : 
F DEBUG   : backtrace:
F DEBUG   :     #01 pc 0000000000218188  (tquic::endpoint::Endpoint::process_connections::hae234b739833ebc9+2384)
F DEBUG   :     #02 pc 0000000000211d08  (quic_endpoint_process_connections+12)

以上为堆栈信息.  平台是Android 9 （ARM 64） ，tquic 版本是：v0.8.0

**Comment by glime-ai** [comment]
根据您提供的堆栈信息，TQUIC在Android 9（ARM 64）平台上运行时遇到了SIGBUS错误。这个错误通常与内存对齐问题有关。以下是一些可能的解决方案：

1. **检查内存对齐**：确保所有内存访问都是正确对齐的。Rust通常会处理这些问题，但在与C代码交互时可能会出现问题。

2. **更新TQUIC版本**：您使用的是TQUIC v0.8.0，建议尝试更新到最新版本，看看问题是否已经被修复。

3. **编译选项**：确保在编译时使用了正确的编译选项。特别是对于Android平台，可能需要特定的编译选项来确保内存对齐。

4. **调试信息**：增加更多的调试信息，特别是在`Endpoint::process_connections`函数中，以便更好地了解问题发生的具体位置和原因。

5. **参考文档和社区支持**：查看TQUIC的官方文档和社区支持，可能会有类似问题的解决方案。例如，您可以在[GitHub Discussions](https://github.com/Tencent/tquic/discussions)中寻找相关讨论。

如果问题仍然存在，建议提供更多的上下文信息或代码片段，以便进一步诊断。

This comment was generated by [Glime](https://glimelab.ai/?utm_source=44520fce49ba43798485a628d6928e22&utm_medium=github).

**Comment by iyangsj** [comment]
@xiaofei0800 看一看

**Comment by xiaofei0800** [comment]
与最新版本差异较大，建议先升级到TQUIC最新版本
